docs: Add release notes for Metals 1.4.2 (#7030)
scalameta/metals: website/blog/2024-12-13-palladium.md

    <td align="center">93</td>
  </tr>
  <tr>
    <td>Merged PRs</td>
    <td align="center">67</td>
  </tr>
    <tr>
    <td>Contributors</td>
    <td align="center">12</td>
  </tr>
  <tr>
    <td>Closed issues</td>
    <td align="center">17</td>
  </tr>
  <tr>
    <td>New features</td>
    <td align="center">1</td>
  </tr>
</tbody>
</table>

For full details:
[https://github.com/scalameta/metals/milestone/74?closed=1](https://github.com/scalameta/metals/milestone/74?closed=1)

Metals is a language server for Scala that works with VS Code, Vim, Emacs, Zed,
Helix and Sublime Text. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from contributors from the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

- [Infer method code action](#infer-method-code-action)

## Infer method code action

Thanks to great work by [ag91](https://github.com/ag91) it's now possible to ask
Metals to infer a method definition based on it's usage when symbol was not
found.

This works similar to the "Create new symbol" code action which would create a

> new Scala file with the added symbol, however in this case for anything starting
> with a lower case we offer to create a method instead.

Reviewer: ```suggestion
new Scala file with the added symbol. Now, for anything starting
with a lower case we offer to create a method instead.
```

This will work both when the symbol is used as a parameter:

![first-example](https://i.imgur.com/9fWQ3Yg.gif)

> as well as standalone function call:

Reviewer: ```suggestion
as well as a standalone function call:
```
